src/analysis/normed_space/exponential.lean

```diff
@@ -319,22 +372,90 @@ lemma exp_analytic (x : 𝔸) :
   analytic_at 𝕂 (exp 𝕂 𝔸) x :=
 analytic_at_exp_of_mem_ball x ((exp_series_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _)
 
-end complete_algebra
-
 /-- In a Banach-algebra `𝔸` over `𝕂 = ℝ` or `𝕂 = ℂ`, if `x` and `y` commute, then
 `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`. -/
-lemma exp_add_of_commute [complete_space 𝔸]
+lemma exp_add_of_commute
   {x y : 𝔸} (hxy : commute x y) :
   exp 𝕂 𝔸 (x + y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y) :=
 exp_add_of_commute_of_mem_ball hxy ((exp_series_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _)
   ((exp_series_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _)
 
+section
+variables (𝕂)
+
+/-- `exp 𝕂 𝔸 x` has explicit two-sided inverse `exp 𝕂 𝔸 (-x)`. -/
+noncomputable def invertible_exp [char_zero 𝕂] (x : 𝔸) : invertible (exp 𝕂 𝔸 x) :=
+invertible_exp_of_mem_ball $ (exp_series_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _
```

Comment:
```suggestion
noncomputable def invertible_exp (x : 𝔸) : invertible (exp 𝕂 𝔸 x) :=
invertible_exp_of_mem_ball $ (exp_series_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _
```
Same with the rest of this section, we have this instance for `is_R_or_C 𝕂`. Also, how about making this an instance and removing this awkward requirement from `inv_of_exp`?

Reply:
It seems we don't tend to make `invertible` declarations actually `instance`s, as this easily leads to diamonds. The awkward requirement in `inv_of_exp` is deliberate, as it makes this lemma immune to diamond problems.

Reply:
Fair enough. Let's just remove the `[char_zero 𝕂]` assumptions and merge this.

Reply:
Yep, I agree I should do that.